src/core/net/ip6_address.hpp

```diff
@@ -484,66 +532,59 @@ class Address : public otIp6Address, public Equatable<Address>, public Clearable
     }
 
     /**
-     * This method returns a pointer to the Interface Identifier.
+     * This method returns the Interface Identifier of the IPv6 address.
      *
-     * @returns A pointer to the Interface Identifier.
+     * @returns A reference to the Interface Identifier.
      *
      */
-    const uint8_t *GetIid(void) const { return mFields.m8 + kInterfaceIdentifierOffset; }
-
-    /**
-     * This method returns a pointer to the Interface Identifier.
-     *
-     * @returns A pointer to the Interface Identifier.
-     *
-     */
-    uint8_t *GetIid(void) { return mFields.m8 + kInterfaceIdentifierOffset; }
+    const InterfaceIdentifier &GetIid(void) const
+    {
+        return static_cast<const InterfaceIdentifier &>(mFields.mComponents.mIid);
+    }
 
     /**
-     * This method indicates whether or not the IPv6 address has the specified Interface Identifier.
+     * This method returns the Interface Identifier of the IPv6 address.
      *
-     * @param[in]  aIid  A pointer to the Interface Identifier.
-     *
-     * @retval true  If the IPv6 address has the specified Interface Identifier.
-     * @retval false If the IPv6 address doesn't have the specified Interface Identifier.
+     * @returns A reference to the Interface Identifier.
      *
      */
-    bool HasIid(const uint8_t *aIid) const;
+    InterfaceIdentifier &GetIid(void) { return static_cast<InterfaceIdentifier &>(mFields.mComponents.mIid); }
 
     /**
      * This method indicates whether or not the IPv6 address has the specified Interface Identifier.
      *
-     * @param[in]  aIid  A reference to the Interface Identifier.
+     * @param[in]  aIid  An Interface Identifier.
      *
      * @retval true  If the IPv6 address has the specified Interface Identifier.
      * @retval false If the IPv6 address doesn't have the specified Interface Identifier.
      *
      */
-    bool HasIid(const InterfaceIdentifier &aIid) const { return HasIid(aIid.m8); }
+    bool HasIid(const InterfaceIdentifier &aIid) const { return GetIid() == aIid; }
 
     /**
      * This method sets the Interface Identifier.
      *
-     * @param[in]  aIid  A pointer to the Interface Identifier.
+     * @param[in]  aIid  An Interface Identifier.
      *
      */
-    void SetIid(const uint8_t *aIid);
+    void SetIid(const InterfaceIdentifier &aIid) { GetIid() = aIid; }
 
     /**
-     * This method sets the Interface Identifier.
+     * This method sets the Interface Identifier from a given byte array.
      *
-     * @param[in]  aIid  A reference to the Interface Identifier.
+     * @param[in] aBuffer    Pointer to an array containing the Interface Identifier. `InterfaceIdentifier::kSize`
+     *                       bytes from the buffer are copied to form the Interface Identifier.
      *
      */
-    void SetIid(const InterfaceIdentifier &aIid) { SetIid(aIid.m8); }
+    void SetIidFromByteArray(const uint8_t *aIidBuffer) { GetIid().SetFromByteArray(aIidBuffer); }
```

Comment:
nit: Can we remove these `SetIidFrom...` methods because typing `GetIid().SetFromByteArray(aIidBuffer)` is not much more difficult than typing `SetIidFromByteArray(aIidBuffer)`?

Reply:
Nice idea. I agree it may be cleaner to not repeat the same methods over in the `Ip6::Address`

Reply:
Updated this in the new pushed commit. 
Also moved more of the methods related to IID from `Address` to `InterfaceIdentifier` class.